### PR TITLE
[null-safety] migrate package:process to _almost_ null safety

### PR DESCRIPTION
**Pull request by jonahwilliams**
Migrate to null-safety, except for the path/file/platform dependencies which are not yet migrated.
Removes record/replay functionality.

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -5,12 +5,11 @@ homepage: https://github.com/google/process.dart
...
-  meta: ^1.1.2
   path: ^1.5.1
   platform: '>=1.0.1'
 
 dev_dependencies:
   test: ^1.0.0
 
 environment:
-  sdk: '>=2.0.0 <3.0.0'
+  sdk: '>=2.9.0 <3.0.0'
```
Comment (tvolkert) [inline pubspec.yaml:15]: Not `2.10.0`?

File: lib/src/interface/process_wrapper.dart [hunk 2]
```diff
@@ -53,15 +54,14 @@ class ProcessWrapper implements io.Process {
   ///
   /// The future returned here will complete with the exit code of the process.
   Future<int> get done async {
-    int result;
+    late int result;
```
Comment (tvolkert) [inline lib/src/interface/process_wrapper.dart:57]: I love this feature.

**Review by tvolkert — APPROVED** [review]

File: pubspec.yaml (hunk 1 above)
Comment (jonahwilliams) [reply pubspec.yaml:15]: Ahh right, we switched versions. 😄

Comment (jonahwilliams) [reply pubspec.yaml:15]: actually we're currently at 2.9 and the next version is null safety. I think flutter is on an allowlist to specify a 2.10 constraint which this package isn't on yet

Comment (tvolkert) [reply pubspec.yaml:15]: ack

**Review by tvolkert — APPROVED** [review]

**Comment by bartekpacia** [comment]
Why was record/replay removed? README still advertises the package as having this feature.

**Pull request merged**